libzeth/core/utils.hpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#ifndef __ZETH_UTIL_HPP__
```

Comment:
`__ZETH_UTIL_HPP__` -> `__ZETH_CORE_UTILS_HPP__`

Reply:
I was going to mention this later, but it may be sensible now: are you happy to move to `#pragma once` for all include instead of these traditional include guards?  I could do that as part of this PR to fix all these I forgot to change.

Reply:
I saw that you used `#pragma once` is some files which breaks consistency with the rest of the code base. Using `#pragma once` is appealing: it is less code to write (and maintain - which makes this type of refactoring easier), and is nicer. However, I read some time ago that it wasn't yet included in the standard (although widely supported), and that it could be problematic in some edge cases. I don't know if the situation as changed but as of now I would stick to the include guards.

Reply:
I'll keep the traditional guards, then.

(AFAIK, the situation is that it's still not "standard", in the sense of being in the C++ standards, but supported by all major compilers for a while now).

Reply:
Right. Feel free to open a ticket to flag the possibility to switch to `#pragma once` in a second step. I am not fully opposed to that, but I would like to understand better the consequences of switching to that (especially the edge cases in which this can be problematic)